tests/realtikvtest/addindextest4/integration_test.go

```diff
@@ -99,3 +103,53 @@ func TestMultiSchemaChangeTwoIndexes(t *testing.T) {
 		tk.MustExec("admin check table t;")
 	}
 }
+
+func TestAdminAlterAddIndexLocalIngestDDLJob(t *testing.T) {
+	store := realtikvtest.CreateMockStoreAndSetup(t)
+	tk1 := testkit.NewTestKit(t, store)
+	tk1.MustExec("use test")
+	tk1.MustExec("create table t (a int);")
+	tk1.MustExec("insert into t values (1);")
+	tk1.MustExec("set @@global.tidb_enable_dist_task=off;")
+	ch := make(chan struct{})
+	testfailpoint.EnableCall(t, "github.com/pingcap/tidb/pkg/ddl/mockIndexIngestWorkerStuck", func() {
+		<-ch
+	})
+
+	testfailpoint.Enable(t, "github.com/pingcap/tidb/pkg/ddl/updateProgressIntervalInMs", "return(100)")
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		tk1.MustExec("alter table t add index idx_a(a);")
+	}()
+	realWorkerCnt := 0
+	realBatchSize := 0
+	realMaxWriteSpeed := 0
+	testfailpoint.EnableCall(t, "github.com/pingcap/tidb/pkg/ddl/checkReorgConcurrency", func(j *model.Job) {
+		realWorkerCnt = j.ReorgMeta.GetConcurrency()
+		realBatchSize = j.ReorgMeta.GetBatchSize()
+		realMaxWriteSpeed = j.ReorgMeta.GetMaxWriteSpeed()
+	})
+
+	jobID := ""
+	tk2 := testkit.NewTestKit(t, store)
+	for {
+		row := tk2.MustQuery("select job_id from mysql.tidb_ddl_job").Rows()
+		if len(row) == 1 {
+			jobID = row[0][0].(string)
+			break
+		}
+	}
+	workerCnt := 7
+	batchSize := 89
+	maxWriteSpeed := 1011
+	tk2.MustExec(fmt.Sprintf("admin alter ddl jobs %s thread = %d", jobID, workerCnt))
+	tk2.MustExec(fmt.Sprintf("admin alter ddl jobs %s batch_size = %d", jobID, batchSize))
+	tk2.MustExec(fmt.Sprintf("admin alter ddl jobs %s max_write_speed = %d", jobID, maxWriteSpeed))
+	require.Eventually(t, func() bool {
+		return realWorkerCnt == workerCnt && realBatchSize == batchSize && realMaxWriteSpeed == maxWriteSpeed
+	}, time.Second*5, time.Millisecond*100)
```

Comment:
```suggestion
	}, 30*time.Second, time.Millisecond*100)
```
to avoid CI slowness and cause flaky, and also change others too